### PR TITLE
Module/cutesv/1.0

### PR DESCRIPTION
**Pull request by nthomas11**
# Pull Request Checklists

**Important:** When opening a pull request, keep only the applicable checklist and delete all other sections.

### Required

- [x] I used the cookiecutter template and updated the placeholder rules.

- [x] The snakemake rules follow the [design guidelines](https://lcr-modules.readthedocs.io/en/latest/for_developers.html#module-rules). 

  - [x] All references to the `rules` object (_e.g._ for input files) are wrapped with `str()`.

- [x] Every rule in the module is either listed under `localrules` or has the `threads` and `resources` directives.

- [x] Input and output files are being symlinked into the `CFG["inputs"]` and `CFG["outputs"]` subdirectories, respectively.

- [x] I grouped the input symlinking rule to the next job that uses the input files. 

- [x] I updated the final target rule (`*_all`) to include every output rule.

- [x] I explained important module design decisions in `CHANGELOG.md`.

- [x] I tested the module on real data for all supported `seq_type` values.

- [x] I updated the `default.yaml` configuration file to provide default values for each rule in the module snakefile.

- [x] I did not set any global wildcard constraints. Any/all wildcard constraints are set on a per-rule basis. 

- [x] I ensured that all symbolic links are relative and self-contained (_i.e._ do not point outside of the repository).

- [x] I replaced every value that should (or might need to) be updated in the default configuration file with `__UPDATE__`.

- [x] I recursively searched for all comments containing `TODO` to ensure none were left. For example:

  ```bash
  grep -r TODO modules/<module_name>/1.0
  ```

### If applicable

- [x] I added more granular output subdirectories.

- [x] I added rules to the `reference_files` workflow to generate any new reference files.

- [x] I added subdirectories with large intermediate files to the list of `scratch_subdirectories` in the `default.yaml` configuration file.

- [x] I updated the list of available wildcards for the input files in the `default.yaml` configuration file.

## Checklist for Updated Module

Important! If you are updating the module version, ensure the previous version of the module is restored from master.
If you want to restore a deleted file or directory from the remote master, you can use `git checkout origin/master path/to/file`,
then a `git commit` will ensure that file is tracked on your branch again.
Example:
```
mv modules/strelka/1.1 modules/strelka/1.2
git checkout origin/master modules/strelka/1.1
```

### CODE REVIEW

File: modules/cutesv/1.0/cutesv.smk [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+    conda :
+        CFG["conda_envs"]["cutesv"] 
+    log:
+        stderr = CFG["logs"]["cutesv"] + "{seq_type}--{genome_build}/{sample_id}/cutesv.stderr.log"             
+    shell:
+         op.as_one_line("""
+            mkdir {output.dir} &&
+            cuteSV {input.bam} {input.fasta} {output.vcf} {output.dir}
+            --threads {threads} --max_cluster_bias_INS 100 --diff_ratio_merging_INS 0.3 
+            --max_cluster_bias_DEL 100 --diff_ratio_merging_DEL 0.3""")
```
Comment (lkhilton) [inline modules/cutesv/1.0/cutesv.smk]: Would users ever want/need to adjust these values, and if yes can you add them to the config and parse them under params?

**Review by lkhilton — CHANGES_REQUESTED** [review]
Looks good - see some comments below. Make sure your log files are set up to capture the stderr for all your rules!

File: modules/cutesv/1.0/cutesv.smk [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+
+# Symlinks the final output files into the module results directory (under '99-outputs/')
+rule _cutesv_output:
+    input:
+        cutesv = str(rules._cutesv.output.vcf)
+    output:
+        cutesv = CFG["dirs"]["outputs"] + "cutesv/{seq_type}--{genome_build}/{sample_id}.vcf"
+
+    run:
+        op.relative_symlink(input.cutesv, output.cutesv, in_module= True)
```
Comment (lkhilton) [inline modules/cutesv/1.0/cutesv.smk]: Same here as for Sniffles - can you add a bedpe-generating step?

Comment (nthomas11) [reply modules/cutesv/1.0/cutesv.smk]: Added rule for vcftobedpe

File: modules/cutesv/1.0/cutesv.smk (hunk 1 above)
Comment (nthomas11) [reply modules/cutesv/1.0/cutesv.smk]: Updated rule to place values in params so user can specify if they want to change the values

File: modules/cutesv/1.0/config/default.yaml [hunk 3]
```diff
@@ -6,17 +6,26 @@ lcr-modules:
             sample_bam: "__UPDATE__"
             sample_bai: "__UPDATE__"
         
-        scratch_subdirectories: []
+        scratch_subdirectories: ["cutesv_working"]
```
Comment (lkhilton) [inline modules/cutesv/1.0/config/default.yaml]: Could you please leave this blank for the default? It can cause some issues on re-runs of the same module.

**Review by lkhilton — CHANGES_REQUESTED** [review]
Looks great, thank you for adding the bedpe step. Once you revert the default `scratch_subdirectories` value I will approve!

File: modules/cutesv/1.0/config/default.yaml (hunk 3 above)
Comment (nthomas11) [reply modules/cutesv/1.0/config/default.yaml]: Fixed!

**Review by lkhilton — APPROVED** [review]

**Pull request merged**